alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,529 @@
+				hostGroup.SysDiskCapacity = v.(string)
+			}
+
+			if v, ok := kv["disk_capacity"]; ok {
+				hostGroup.DiskCapacity = v.(string)
+			}
+
+			if v, ok := kv["sys_disk_type"]; ok {
+				hostGroup.SysDiskType = v.(string)
+			}
+
+			if v, ok := kv["disk_type"]; ok {
+				hostGroup.DiskType = v.(string)
+			}
+
+			if v, ok := kv["host_group_name"]; ok {
+				hostGroup.HostGroupName = v.(string)
+			}
+
+			if v, ok := kv["disk_count"]; ok {
+				hostGroup.DiskCount = v.(string)
+			}
+
+			if v, ok := kv["auto_renew"]; ok {
+				if v.(bool) == false {
+					hostGroup.AutoRenew = "false"
+				} else if v.(bool) == true {
+					hostGroup.AutoRenew = "true"
+				}
+			}
+
+			if v, ok := kv["gpu_driver"]; ok {
+				hostGroup.GpuDriver = v.(string)
+			}
+
+			if v, ok := kv["node_count"]; ok {
+				hostGroup.NodeCount = v.(string)
+			}
+
+			if v, ok := kv["instance_type"]; ok {
+				hostGroup.InstanceType = v.(string)
+			}
+
+			if v, ok := kv["charge_type"]; ok {
+				hostGroup.ChargeType = v.(string)
+			}
+
+			if v, ok := kv["host_group_type"]; ok {
+				hostGroup.HostGroupType = v.(string)
+			}
+
+			hostGroups = append(hostGroups, hostGroup)
+		}
+	}
+	request.HostGroup = &hostGroups
+
+	raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+		return emrClient.CreateClusterV2(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*emr.CreateClusterV2Response)
+	d.SetId(response.ClusterId)
+	return nil
+}
+
+func resourceAlicloudEmrClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	emrService := EmrService{client}
+
+	object, err := emrService.DescribeEmrCluster(d.Id())
+
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	d.Set("name", object.ClusterInfo.Name)
+	d.Set("charge_type", object.ClusterInfo.ChargeType)
+
+	d.Set("high_availability_enable", object.ClusterInfo.HighAvailabilityEnable)
+	d.Set("net_type", object.ClusterInfo.NetType)
+	d.Set("security_group_id", object.ClusterInfo.SecurityGroupId)
+	d.Set("net_type", object.ClusterInfo.NetType)
+	d.Set("vpc_id", object.ClusterInfo.VpcId)
+	d.Set("vswitch_id", object.ClusterInfo.VSwitchId)
+	d.Set("use_local_metadb", object.ClusterInfo.UseLocalMetaDb)
+	d.Set("deposit_type", object.ClusterInfo.DepositType)
+	d.Set("eas_enable", object.ClusterInfo.EasEnable)
+	d.Set("user_defined_emr_ecs_role", object.ClusterInfo.UserDefinedEmrEcsRole)
+	d.Set("related_cluster_id", object.ClusterInfo.RelateClusterId)
+
+	return nil
+}
+
+func resourceAlicloudEmrClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	d.Partial(true)
+	if d.HasChange("name") {
+		d.SetPartial("name")
```

Comment:
Please move the line 412 to 419. This action should be run after invoking API request.

Reply:
fixed